courses/models_test.py

```diff
@@ -148,6 +148,33 @@ def test_course_run_past(end_days, expected):
     assert CourseRunFactory.create(end_date=end_date).is_past is expected
 
 
+@pytest.mark.parametrize(
+    "start_delta, end_delta, expiration_delta, exceptionExpected",
+    [[-1, 2, 3, False], [1, 2, 1, True], [1, 2, -1, True]],
+)
+def test_course_run_expiration_date(
+    start_delta, end_delta, expiration_delta, exceptionExpected
+):
+    """
+    Test that CourseRun.expiration_date returns the expected value and raises ValidationError
+    if expiration_date is before start_date or end_date
+    """
+    now = now_in_utc()
+    if exceptionExpected:
+        with pytest.raises(ValidationError):
+            CourseRunFactory.create(
+                start_date=now + timedelta(days=start_delta),
+                end_date=now + timedelta(days=end_delta),
+                expiration_date=now + timedelta(days=expiration_delta),
+            )
+    else:
```

Comment:
I would recommend to divide this tests into two different tests for happy and sad paths. Conditional statements are not recommended in tests.

Reply:
Broke it down into separate tests.